### PR TITLE
Add beam manager

### PR DESCRIPTION
**Pull request by iwko**

### CODE REVIEW

File: test/test_beam_manager.py [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+class DataFlowManagerTest(BeamTestCase):
+    def setUp(self):
+        super(DataFlowManagerTest, self).setUp()
+        self.data_flow_manager = create_dataflow_manager(
+            project_id='project',
+            runtime="2019-01-01",
+            dataset_name='test_dataset',
+            dataflow_bucket="bucket",
+            internal_tables={"internal_table": "project.dataset.internal_table"},
```
Comment (beczkowb) [inline test/test_beam_manager.py]: internal_tables should be a list of table names(manager can construct a full table id from given project_id and dataset_name.

File: biggerquery/beam_manager.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+            shard_name_template='S_N',
+            file_name_suffix='.avro')
+
+
+class TemplatedBeamManager:
+    """
+    Decorator that resolves table names/all kind of variables.
+    """
+
+    def __init__(self, beam_manager, internal_tables, external_tables, run_datetime, extras):
```
Comment (beczkowb) [inline biggerquery/beam_manager.py]: change run_datetime to runtime everywhere

File: biggerquery/beam_manager.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        self.beam_manager = beam_manager
+        self.extras = extras
+        self.internal_tables = internal_tables
+        self.external_tables = external_tables
+        self.run_datetime = run_datetime
+
+    def read_from_big_query(self, dataset, project_id, standard_sql, query):
```
Comment (beczkowb) [inline biggerquery/beam_manager.py]: I think that sql is enough, we support only standard sql

File: biggerquery/beam_manager.py [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+            table_name,
+            schema=schema,
+            write_disposition=BigQueryDisposition.WRITE_TRUNCATE)
+
+    def _legacy_table_name(self, table_name):
+        spl = table_name.split('.')
+        return spl[0] + ':' + '.'.join(spl[1:len(spl)])
+
+    @property
+    def run_datetime(self):
```
Comment (beczkowb) [inline biggerquery/beam_manager.py]: run_datetime -> run_datetime_str, I know its bad, but let's be consistent with the bigquery dataset manager

File: biggerquery/beam_manager.py [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+            options.view_as(StandardOptions).runner = 'DataflowRunner'
+        else:
+            options.view_as(StandardOptions).runner = 'DirectRunner'
+        self.beam_manager.beam_manager.pipeline = beam.Pipeline(options=options)
+        return self.beam_manager.beam_manager.pipeline
+
+    def _job_name(self, day, job_name):
+        if 'job_suffix' in self.beam_manager.extras:
+            import re
+            table_prefix_slug = re.sub(r'[^a-z0-9]+', '-', self.beam_manager.extras['job_suffix']).strip('-')
```
Comment (beczkowb) [inline biggerquery/beam_manager.py]: get rid of table_prefix

File: biggerquery/beam_manager.py [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+def _throw_on_none(param, param_name):
+    if param is None:
+        raise ValueError("%s must be defined" % param_name)
+
+
+def create_dataflow_manager(
+        project_id,
+        runtime,
+        dataset_name,
+        dataflow_bucket,
```
Comment (beczkowb) [inline biggerquery/beam_manager.py]: internal_tables and external_tables are optional so they should be None by default. I think that the machine_type parameter should has default value to(standard machine).

Comment (iwko) [reply biggerquery/beam_manager.py]: external_tables is optional, right. But without internal_tables it doesnt make sense to use this tool

File: test/test_beam_manager.py (hunk 1 above)
Comment (iwko) [reply test/test_beam_manager.py]: Look now please.

File: biggerquery/beam_manager.py (hunk 2 above)
Comment (iwko) [reply biggerquery/beam_manager.py]: Look now please.

File: biggerquery/beam_manager.py (hunk 3 above)
Comment (iwko) [reply biggerquery/beam_manager.py]: Look now please.

File: biggerquery/beam_manager.py (hunk 4 above)
Comment (iwko) [reply biggerquery/beam_manager.py]: Look now please.

File: biggerquery/beam_manager.py (hunk 5 above)
Comment (iwko) [reply biggerquery/beam_manager.py]: Look now please.

File: biggerquery/beam_manager.py (hunk 6 above)
Comment (iwko) [reply biggerquery/beam_manager.py]: Look now please.

Comment (beczkowb) [reply biggerquery/beam_manager.py]: You can write full table names without using aliases, so it make sense.

Comment (iwko) [reply biggerquery/beam_manager.py]: Oh, right

Comment (iwko) [reply biggerquery/beam_manager.py]: Look now please

Comment (iwko) [reply biggerquery/beam_manager.py]: Look now please

**Pull request merged**